### ISSUE TITLE
Can't understand build

### ISSUE THREAD

**Issue by JorgePerC** [micro-ROS/micro_ros_stm32cubemx_utils#121]
Hey, newcomer here. 
I'm trying to incorporate this into a new CubeIDE project, but can't seem to understand how to do it. Specially with steps 3 and 4, I clone this repo into my project, yet, I fail at _building?_ libmicroros. I guess this is done through docker, but then how can I find the absolute path at step 4?
I've tried to run the library_generation.sh in my local machine but fail at doing so. 
I've completed the tutorial at https://micro.ros.org/docs/tutorials/core/first_application_linux/

**Comment by pablogs9** [comment]
Hello, we need more details about your failure. In which step it fails?

You mention problems with steps 3 and 4:

![image](https://github.com/micro-ROS/micro_ros_stm32cubemx_utils/assets/7647884/d42180c2-df6b-4e32-9ff5-8e40efe4e71b)

Could you elaborate on this?

**Comment by JorgePerC** [comment]
So... I had an initial error due to working on a board with two micros.
Tried again with a single one,  and copypasted the paths as the instructions said. 
//question: should step 3 be repeated for MCU GCC and MCU G++ compiler?


![Screenshot from 2023-10-11 14-05-50](https://github.com/micro-ROS/micro_ros_stm32cubemx_utils/assets/42452493/0d34d543-0930-45dd-aeed-082972bfbe29)

----------
It eventually got to the build process. However, it failed. It is a missing file _<uxr/client/transport.h>_
![Screenshot from 2023-10-11 14-51-04](https://github.com/micro-ROS/micro_ros_stm32cubemx_utils/assets/42452493/59c089d6-0a3e-4974-86b2-cddced0d9b67)


Then I found a similar [issue](https://github.com/micro-ROS/micro_ros_stm32cubemx_utils/issues/113)
But I got question, if I want to modify it, should I just edit the file locally on the clone of this repo? or is it dependent on the Docker image?

Btw, thanks for the help :D

**Comment by pablogs9** [comment]
Yes, if you are using C and C++ you will need to add the include paths for both compilers
